### PR TITLE
Make LockAcquiringTimeoutException a RuntimeException

### LINKED ISSUE #16690: Make LockAcquiringTimeoutException a runtime exception
### Description

With the refactoring of the global lock in #9388, there is a new exception `LockAcquiringTimeoutException` that extends `Exception`. When implementing a `RuntimeException`, this would lead to less code.

### Motivation

Runtime exceptions don't require declaration in methods, and don't need to be caught unless one want to handle them. 

The current pattern in the code is to catch them and wrap them as a generic `RuntimeException`: 

https://github.com/keycloak/keycloak/blob/6566b58be177c126cac30ed5e5196e4ab5555cb0/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/database/LegacyJpaConnectionProviderFactory.java#L298-L300

### PR DESCRIPTION
**Pull request by ahus1**
Closes #16690

### CODE REVIEW

**Comment by ahus1** [comment]
@mhajas - this is now ready for review; the code with `RuntimeException` is now shorter than before. Thanks!

File: server-spi-private/src/main/java/org/keycloak/models/locking/LockAcquiringTimeoutException.java [hunk 1]
```diff
@@ -22,9 +22,7 @@
 /**
  * This exception is thrown when acquiring a lock times out.
  */
-public final class LockAcquiringTimeoutException extends Exception {
-
-    private LockAcquiringTimeoutException() {}
+public final class LockAcquiringTimeoutException extends RuntimeException {
```
Comment (mhajas) [inline server-spi-private/src/main/java/org/keycloak/models/locking/LockAcquiringTimeoutException.java:25]: Should we return the non-parametric constructor? This is to make sure the values `String lockName, String keycloakInstanceIdentifier, Instant timeWhenAcquired` are always set

**Review by mhajas — CHANGES_REQUESTED** [review]

File: server-spi-private/src/main/java/org/keycloak/models/locking/LockAcquiringTimeoutException.java (hunk 1 above)
Comment (mhajas) [reply server-spi-private/src/main/java/org/keycloak/models/locking/LockAcquiringTimeoutException.java:25]: You are right. Sorry, it turned out I forgot how the no-argument constructor behaves when there is some parameteric added :man_facepalming:

**Review by mhajas — APPROVED** [review]
LGTM thanks!

**Pull request merged**